CoreScriptsRoot/ServerCoreScripts/ServerChat/Speaker.lua

```diff
@@ -0,0 +1,187 @@
+local source = [[
+local module = {}
+
+--////////////////////////////// Details
+--//////////////////////////////////////
+local metatable = {}
+metatable.__ClassName = "Speaker"
+
+metatable.__tostring = function(tbl)
```

Comment:
I would personally opt for a simpler approach in the form of this kind of pattern:

local YourClass = {}
YourClass.__index = YourClass
function YourClass.new()
     local self = setmetatable({}, YourClass)
     self.member = 5
     return self
end

function YourClass:__tostring() ... end
function YourClass:SomeMethod() ... end

That way there's not really much room to wonder what it does and you get your metamethods and normal methods all in one place. Checking the type is just checking getmetatable(instance)==YourClass.